### ISSUE TITLE
Update data access examples to use sqlclient

### ISSUE THREAD

**Issue by rdruss** [vert-x3/vertx-guide-for-java-devs#78]
This is a suggestion/question.  Given the introduction of asynchronous data access through the sqlclient, should this be the recommended method for doing data access as documented in the java guide?  @jponge, wdyt?

**Comment by jponge** [comment]
Absolutely, the whole guide needs to be revamped for v4 (e.g., no more Shiro authentication, move to SQL client, etc). And perhaps make the guide more concise.
